src/main/java/com/mixpanel/android/mpmetrics/MPDbAdapter.java

```diff
@@ -179,13 +243,19 @@ public int addJSON(JSONObject j, Table table) {
      * Removes events with an _id <= last_id from table
      * @param last_id the last id to delete
      * @param table the table to remove events from, either "events" or "people"
+     * @param includeAutomaticEvents whether or not automatic events should be included in the cleanup
      */
-    public void cleanupEvents(String last_id, Table table) {
+    public void cleanupEvents(String last_id, Table table, String token, boolean includeAutomaticEvents) {
         final String tableName = table.getName();
 
         try {
             final SQLiteDatabase db = mDb.getWritableDatabase();
-            db.delete(tableName, "_id <= " + last_id, null);
+            StringBuffer deleteQuery = new StringBuffer("_id <= " + last_id + " AND " + KEY_TOKEN + " = '" + token + "'");
+
+            if (!includeAutomaticEvents) {
+                deleteQuery.append(" AND " + KEY_AUTOMATIC_DATA + "=0");
+            }
+            db.delete(tableName, deleteQuery.toString(), null);
```

Comment:
if `includeAutomaticEvents` is false then we don't delete the automatic events! (`=0`)!